### PR TITLE
adding some code

### PR DESCRIPTION
**Pull request by olhabn**
so it looked like we needed to match front-end keyword to back-end keyword or otherwise on search  we find record with backend Keyword but we don't find a record with frontend Keyword on dblclick.
so lines 2422-2432 correspond to line 2478.

### CODE REVIEW

File: src/foam/mlang/mlang.js [hunk 1]
```diff
@@ -2418,10 +2418,18 @@ foam.CLASS({
...
             } else {
-              s = prop.f(obj).toLowerCase();
+              s = prop.f(obj);
+
+              if ( ! ( typeof s === "string" ) ) {
+                if ( s.toSummary ) {
+                  s = s.toSummary();
+                } else if ( s.toString ) {
+                  s = s.toString();
+                } else {
```
Comment (lchanmann) [inline src/foam/mlang/mlang.js]: I don't think keyword should test toSummary() and toString()

File: src/foam/mlang/mlang.js [hunk 2]
```diff
@@ -2418,10 +2418,18 @@ foam.CLASS({
               s = prop.f(obj).toString().toLowerCase();
             } else if ( this.Date.isInstance(prop) ) {
               s = prop.f(obj).toISOString().toLowerCase();
-            } else if ( ! this.String.isInstance(prop) ) {
-              continue;
             } else {
-              s = prop.f(obj).toLowerCase();
+              s = prop.f(obj);
+
+              if ( ! ( typeof s === "string" ) ) {
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js]: foam.String.isInstance(s) is better

File: src/foam/mlang/mlang.js (hunk 1 above)
Comment (olhabn) [reply src/foam/mlang/mlang.js]: fixed

File: src/foam/mlang/mlang.js (hunk 2 above)
Comment (olhabn) [reply src/foam/mlang/mlang.js]: fixed

File: src/foam/mlang/mlang.js [hunk 3]
```diff
@@ -2418,10 +2418,10 @@ foam.CLASS({
               s = prop.f(obj).toString().toLowerCase();
             } else if ( this.Date.isInstance(prop) ) {
               s = prop.f(obj).toISOString().toLowerCase();
-            } else if ( ! this.String.isInstance(prop) ) {
-              continue;
             } else {
-              s = prop.f(obj).toLowerCase();
+              s = prop.f(obj);
```
Comment (kgrgreer) [inline src/foam/mlang/mlang.js:2422]: Would this be better:
s = String.valueOf(prop.f(obj)).toLowerCase();

Comment (olhabn) [reply src/foam/mlang/mlang.js:2422]: it doesn't work for me for some reason. but i can investigate

**Pull request merged**